### PR TITLE
Sreq

### PR DESCRIPTION
**Pull request by RanjiniM**

### CODE REVIEW

File: src/NuGetGallery/App_Start/AppActivator.cs [hunk 1]
```diff
@@ -135,7 +135,8 @@ private static void BundlingPostStart()
                 .Include("~/Scripts/jquery.validate.unobtrusive.js")
                 .Include("~/Scripts/jquery.timeago.js")
                 .Include("~/Scripts/nugetgallery.js")
-                .Include("~/Scripts/stats.js");
+                .Include("~/Scripts/stats.js")
```
Comment (maartenba) [inline src/NuGetGallery/App_Start/AppActivator.cs]: Do we want to serve all Bootstrap JS and CSS to all users on every request? This adds quite a bit to the average page download size.

Can you make it a separate bundle and only reference it where needed?

File: src/NuGetGallery/App_Start/AppActivator.cs [hunk 2]
```diff
@@ -148,7 +149,8 @@ private static void BundlingPostStart()
             foreach (string filename in new[] {
                     "Site.css",
                     "Layout.css",
-                    "PageStylings.css"
+                    "PageStylings.css",
```
Comment (maartenba) [inline src/NuGetGallery/App_Start/AppActivator.cs]: Do we want to serve all Bootstrap JS and CSS to all users on every request? This adds quite a bit to the average page download size.

Can you make it a separate bundle and only reference it where needed?

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 3]
```diff
@@ -0,0 +1,442 @@
...
+using System.Web;
+using System.Web.Mvc;
+using NuGetGallery.Areas.Admin.Models;
+using NuGetGallery.Areas.Admin.ViewModels;
+
+namespace NuGetGallery.Areas.Admin.Controllers
+{
+    public class RequestController : AdminControllerBase
+    {
+        IssueModel issue = new IssueModel();
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Are these required as private variables? Better to create the model in the action method and then pass the model to the view.

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 4]
```diff
@@ -0,0 +1,442 @@
...
+        AdminModel admin = new AdminModel();
+        HistoryModel history = new HistoryModel();
+
+        IList<IssueViewModel> issueViews = new List<IssueViewModel>();
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Are these required as private variables? Better to create the model in the action method and then pass the model to the view.

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 5]
```diff
@@ -0,0 +1,442 @@
...
+            foreach (var a in allAdmins)
+            {
+                SelectListItem currentItem = new SelectListItem { Text = a.UserName, Value = a.Id.ToString() };
+
+                admins.Add(currentItem);
+            }
+
+            var reasons = CreateListOfReasons(String.Empty);
+
+            ViewBag.IssueStatus = items;
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: ViewBag is "evil" (not strong typed) - can you pass the model as a strong-typed model to the view? E.g. http://www.asp.net/mvc/overview/views/dynamic-v-strongly-typed-views

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 6]
```diff
@@ -0,0 +1,442 @@
...
+            return View();
+        }
+
+        [HttpPost]
+        public ActionResult Create(Issue newissue)
+        {
+
+            if (ModelState.IsValid)
+            {
+                newissue.CreatedDate = DateTime.Now;
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: .UtcNow if possible, server time is always UTC but using UtcNow makes that clear.

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 7]
```diff
@@ -0,0 +1,442 @@
...
+            {
+                return View(newissue);
+            }
+        }
+
+        [HttpPost]
+        public ActionResult Save(FormCollection formcollection)
+        {
+            //Values required for updating the issue
+            var issueId = GetValue(formcollection["IssueId"]);
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: FormCollection is "evil" - none of the elements are strong-typed or provide IntelliSense. Can you convert this to use action method parameters (or a model)?

https://msdn.microsoft.com/en-us/library/dd410269(v=vs.100).aspx#Anchor_4

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 8]
```diff
@@ -0,0 +1,442 @@
...
+            //Values required for remembering the filters applied
+            var reasonFilter = String.Empty;
+
+            if (formcollection["ReasonFilter"] != "/")
+            {
+                reasonFilter = formcollection["ReasonFilter"];
+            }
+
+            //Write values into Session variables
+            Session["StatusId"] = GetValue(formcollection["StatusId"]);
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Why do we need sessions here? Introducing sessions means we also have to introduce additional infrastructure to store session state across our cloud services instances, as not every instance will serve the same end user.

Move away from sessions if possible.

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 9]
```diff
@@ -0,0 +1,442 @@
...
+                rv.OwnerLink = string.Empty;
+                rv.Issue.SiteRoot = VerifyAndFixTralingSlash(rv.Issue.SiteRoot);
+
+                if (!rv.Issue.CreatedBy.Equals("Anonymous", StringComparison.OrdinalIgnoreCase))
+                {
+                    rv.OwnerLink = String.Concat(rv.Issue.SiteRoot, "Profiles", "/",  rv.Issue.CreatedBy);
+                }
+                filteredIssueViews.Add(rv);
+
+                rv.PackageLink = String.Concat(rv.Issue.SiteRoot, "Packages", "/", rv.Issue.PackageID, "/", rv.Issue.PackageVersion);
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Can the view just make use of the UrlHelper/HtmlHelper to generate links? That way the controller does not need to pre-determine each and every link.

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 10]
```diff
@@ -0,0 +1,442 @@
...
+                returnVal = String.Concat(p, "/");
+            }
+            return returnVal;
+        }
+
+        private List<SelectListItem> CreateListOfReasons(string reasonToSelect)
+        {
+            var reasons = new List<SelectListItem>();
+            var item = new SelectListItem { Text = "Other" };
+            reasons.Add(item);
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: `reasons.Add(new SelectListItem { Text = "HasABugOrFailedToInstall" });`

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs [hunk 11]
```diff
@@ -0,0 +1,442 @@
...
+            }
+            else
+            {
+                issues = issue.GetOpenIssues();
+                ViewBag.Title = "Open Support Requests";
+            }
+            return issues;
+        }
+
+        private int GetValue(string input)
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Looking at this method, it's not immediately obvious what it does. Maybe rename to `ConvertToInt` or something like that.

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs (hunk 6 above)
Comment (xavierdecoster) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Unless we care about TimeZoneInfo (I guess not), perhaps prefer `DateTimeOffset.UtcNow`?

File: src/NuGetGallery/Areas/Admin/Models/Admin.cs [hunk 12]
```diff
@@ -0,0 +1,17 @@
+namespace NuGetGallery.Areas.Admin.Models
+{
+    using System;
+    using System.Collections.Generic;
+    using System.ComponentModel.DataAnnotations;
+    using System.ComponentModel.DataAnnotations.Schema;
+    using System.Data.Entity.Spatial;
+
+    public partial class Admin
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Models/Admin.cs]: Does this one map to the users table of the gallery? How do both gallery and support models map to one another?

File: src/NuGetGallery/Areas/Admin/Models/AdminModel.cs [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+        public virtual DbSet<Admin> Admins { get; set; }
+
+        protected override void OnModelCreating(DbModelBuilder modelBuilder)
+        {
+            modelBuilder.Entity<Admin>()
+                .Property(e => e.UserName)
+                .IsUnicode(false);
+        }
+
+        public List<Admin> GetAllAdmins()
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Models/AdminModel.cs]: This type of methods is usually something you want to hide away in a repository and not directly on the DbContext.

File: src/NuGetGallery/Areas/Admin/Models/IssueModel.cs [hunk 14]
```diff
@@ -0,0 +1,155 @@
...
+
+            modelBuilder.Entity<Issue>()
+                .Property(e => e.Reason)
+                .IsUnicode(false);
+        }
+
+        public List<Issue> GetOpenIssues()
+        {
+            var allOpenIssues = from r in Issues
+                                where (r.IssueStatus != 5 && r.IssueStatus != 6)
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Models/IssueModel.cs]: Not very clear what `5` or `6` is, can this become an enum?

File: .vs/config/applicationhost.config [hunk 15]
```diff
@@ -155,7 +155,7 @@
         <sites>
             <site name="NuGet Gallery (nuget.localtest.me)" id="2">
                 <application path="/">
-                    <virtualDirectory path="/" physicalPath="..\..\src\NuGetGallery" />
+                    <virtualDirectory path="/" physicalPath="C:\Users\ranjinim\Documents\GitHub\NuGetGallery\src\NuGetGallery" />
```
Comment (xavierdecoster) [inline .vs/config/applicationhost.config]: won't work on my machine :) do we need applicationhost.config in source repo? perhaps add to `.gitignore`?

File: src/NuGetGallery/Areas/Admin/Views/Request/Index.cshtml [hunk 16]
```diff
@@ -0,0 +1,233 @@
...
+<link href="~/Content/bootstrap.css" rel="stylesheet" />
+<h1>
+    @ViewBag.Title
+</h1>
+
+<p>
+
+</p>
+
+<style>
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Views/Request/Index.cshtml]: Can this go to a .css file?

File: src/NuGetGallery/Areas/Admin/Views/Request/Index.cshtml [hunk 17]
```diff
@@ -0,0 +1,233 @@
+﻿@model IEnumerable<NuGetGallery.Areas.Admin.ViewModels.IssueViewModel>
+
+<link href="~/Content/bootstrap.css" rel="stylesheet" />
+<h1>
+    @ViewBag.Title
+</h1>
+
+<p>
```
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Views/Request/Index.cshtml]: Empty paragraph intended?

**Comment by xavierdecoster** [comment]
Why do we need sessions? I'd prefer the server to be completely stateless, and if need be, persist any state out of process.

File: src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs (hunk 10 above)
Comment (maartenba) [inline src/NuGetGallery/Areas/Admin/Controllers/RequestController.cs]: Maybe make this into Enum's.

**Comment by maartenba** [comment]
Can this PR be closed?

**Pull request closed without merging**